src/app/public/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts

```diff
@@ -1,116 +1,233 @@
 import {
-  Component,
-  Input,
   ChangeDetectionStrategy,
   ChangeDetectorRef,
+  Component,
+  Input,
+  OnDestroy,
   OnInit,
-  OnDestroy
+  Optional
 } from '@angular/core';
 
 import {
-  Subject, BehaviorSubject
-} from 'rxjs';
+  Subject
+} from 'rxjs/Subject';
+
+import 'rxjs/add/operator/distinctUntilChanged';
+
+import 'rxjs/add/operator/takeUntil';
 
 import {
-  SkyLibResourcesService
-} from '@skyux/i18n';
+  SkyProgressIndicatorChange,
+  SkyProgressIndicatorMessageType,
+  SkyProgressIndicatorNavButtonType
+} from '../types';
 
 import {
   SkyProgressIndicatorComponent
 } from '../progress-indicator.component';
-import {
-  SkyProgressIndicatorMessageType,
-  SkyProgressIndicatorChange
-} from '../types';
 
 @Component({
   selector: 'sky-progress-indicator-nav-button',
   templateUrl: './progress-indicator-nav-button.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class SkyProgressIndicatorNavButtonComponent implements OnInit, OnDestroy {
-  @Input()
-  public progressIndicator: SkyProgressIndicatorComponent;
-
   @Input()
   public buttonText: string;
 
   @Input()
-  public get buttonType(): string {
-    return this._buttonType || 'next';
+  public set buttonType(value: SkyProgressIndicatorNavButtonType) {
+    this._buttonType = value;
   }
-  public set buttonType(value: string) {
```

Comment:
Not really, because the two string values that were supported before, are part of a new `string` union type. The input should work exactly the same as it does today.

Reply:
Gotchya. That's cool!

Reply:
The `buttonType` input variable is changing from a `string` to a `SkyProgressIndicatorNavButtonType`. Won't this be considered a breaking change?